### ISSUE TITLE
Phone number - Add country code input to phone number field

### ISSUE THREAD

**Issue by isagoico** [Expensify/App#3798]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Log in to an account that doesn't have a phone number set
2. Navigate to profile settings 
3. Scroll to the phone number field 

## Expected Result:
There should be an input for country codes. It should default to the country code of your current location, and when clicked/tapped, should just be an input for the user to type in their country code. We don't want a big dropdown list of countries, and we don't want to use any additional third-party packages.

## Actual Result:
There's no country code selector or instructions to indicate the user that he must add the country code to the phone number. 

## Workaround:
Unknown 

## Platform:
Where is this issue occurring?

 Web ✔️ 
 iOS✔️
 Android✔️
 Desktop App✔️
 Mobile Web✔️

**Version Number:** 1.0.74-0

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

![image](https://user-images.githubusercontent.com/44479856/123849419-12567d80-d919-11eb-94af-b7d2a6adbe1f.png)

**Expensify/Expensify Issue URL:**
Upwork job post: https://www.upwork.com/jobs/~01ee4f567be50b24b3
[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

___ 
From @aliabbasmalik8 https://expensify.slack.com/archives/C01GTK53T8Q/p1624845554159200
> We can add countries' code dropdown while adding phone numbers because most of the time users did not enter their country code before the number and they have no idea what's wrong with their phone number

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Luke9389 (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by Luke9389** [comment]
Looks like a great External issue 👍

**Comment by MelvinBot** [comment]
Triggered auto assignment to @puneetlath (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by aliabbasmalik8** [comment]
Would like to work on this issue. We can use some components for this issue like `https://www.npmjs.com/package/react-native-phone-number-input` OR can make all work custom.

With a custom approach, we can save all country codes in an array and show them in the dropdown list with phone number input.

Also, it's great to add UI-relevant instruction in the job description.
Thanks

**Comment by ObasiGodstime** [comment]
**Proposal**

It is going to be selected and added to number to be verified like in the video, I am making use of a package that can be customized to add and remove countries (react-native-country-picker-modal)

VIDEO EXAMPLE OF A SOLUTION : -
https://user-images.githubusercontent.com/42383942/124179077-309fb300-daaa-11eb-8d3c-33f7f8f036ef.mp4

**Comment by ObasiGodstime** [comment]
Proposal

It is going to be selected and added to number to be verified like in the video, I am making use of a package that can be customized to add and remove countries (react-native-country-picker-modal)

VIDEO EXAMPLE OF A SOLUTION

https://user-images.githubusercontent.com/42383942/124179483-d3583180-daaa-11eb-9a1e-177b8ee8d459.mp4

: -

**Comment by ObasiGodstime** [comment]
**PROPOSAL**

I have added a country code selector beside the input field for better reach on the user's side, made use of a package(react-native-country-picker-modal) that can be customized to remove and add more countries

VIDEO OF WORK BELOW: -


https://user-images.githubusercontent.com/42383942/124180475-27afe100-daac-11eb-8909-cec136e05f43.mov

**Comment by jonats2020** [comment]
I would like to propose implementing this package.

https://user-images.githubusercontent.com/74584906/124251446-f6302780-db36-11eb-8b7c-57b1d57ba51a.mp4

**Comment by shubhamgrover93** [comment]
Hi, 
Thanks for the invitation. I would like to share a solution for this task. 
We can use a component of React Native (country-picker-modal) with the help of which, the user would be able to select a country. Once a particular country is selected, its country code will be displayed in the given field and then the regular phone digits could be added.

There's an alternative to achieve this with custom-functionality. We can create a list of country codes & save it in the database. And can give that as an option-to-chose for a user along with phone number input.

Please check the attached video for reference:

https://user-images.githubusercontent.com/47270753/124427352-e90a7700-dd88-11eb-9ebe-a27ee3f436a1.mov

Thanks.

**Comment by shubhamgrover93** [comment]
Hi, 
Thanks for the invitation. I would like to share a solution for this task. 
We can use a component of React Native (country-picker-modal) with the help of which, the user would be able to select a country. Once a particular country is selected, its country code will be displayed in the given field and then the regular phone number digits could be added.

There's an alternative to achieve this with custom-functionality. We can create a list of country codes & save it in the database. And can give that as an option for a user along with phone number input.

Please check the attached video:
https://user-images.githubusercontent.com/47270753/124427570-2f5fd600-dd89-11eb-8a36-9e8024fe92f9.mp4

Thanks.

**Comment by ObasiGodstime** [comment]
@shubhamgrover93 did you really have to use my video 😄😄

**Comment by Luke9389** [comment]
Hey contributors, thanks for the proposals thus far. You've really all been saying the same thing, which is that you want to use a package to make this list. We're not gonna go that route. I've updated the description of this issue to indicate that. I'm gonna give @aliabbasmalik8 a day or two to respond because it was his original comment that lead to the creation of this issue.

**Comment by Luke9389** [comment]
I also put this on hold to wait for [this](https://github.com/Expensify/Expensify.cash/pull/3414) to be merged.

**Comment by aliabbasmalik8** [comment]
@Luke9389 would like to work on this when this issue will reopen.

Thanks

**Comment by haseebb** [comment]
Hi 

Yes, I am interested to work with you. 
I checked the issue on Github and I will use this "country picker model" in the app 

https://www.npmjs.com/package/react-native-country-picker-modal

**Comment by Luke9389** [comment]
@haseebb Thank you for your interest, but if you read the above thread you'll see that we're not accepting new proposals for this right now, nor are we interested in using npm packages for this solution.

**Comment by Luke9389** [comment]
I'm going to be OOO for all of next week, so I can get back to this after that

**Comment by Luke9389** [comment]
Quick update, we're still waiting on [this](https://github.com/Expensify/Expensify.cash/pull/3414) issue to get merged before taking this one off HOLD.

**Comment by puneetlath** [comment]
@Luke9389 looks like that has been merged. Shall we take this off hold?

**Comment by Luke9389** [comment]
Oh, nice catch. 👍

**Comment by Luke9389** [comment]
OK, this issue is now off HOLD and accepting new proposals. @aliabbasmalik8 brought this issue to our attention so I'll see if he's still wanting to submit a proposal.

**Comment by aliabbasmalik8** [comment]
**_PROPOSAL_**

We can add an input field here
https://github.com/Expensify/App/blob/f3228deee0ac6a9fabb2c1c5d40ffe31d77dbbd7/src/pages/settings/AddSecondaryLoginPage.js#L128

like this

```
{
    this.formType === CONST.LOGIN_TYPE.PHONE && (
    <View style={styles.mb6}>
        <ExpensiTextInput
            label={this.props.translate('addSecondaryLoginPage.addCountryCode')}
            ref={el => this.countyCodeRef = el}
            value={this.state.countryCode}
            onChangeText={countryCode => this.setState({countryCode})}
            keyboardType={CONST.KEYBOARD_TYPE.PHONE_PAD}
            returnKeyType="done"
        />
    </View>
    )
}
```

and also add `countryCode` in the state, and update submit handler
https://github.com/Expensify/App/blob/f3228deee0ac6a9fabb2c1c5d40ffe31d77dbbd7/src/pages/settings/AddSecondaryLoginPage.js#L87
like this
```
submitForm() {
      const login = `${this.state.countryCode}${this.state.login}`;
      setSecondaryLogin(login, this.state.password)
          .then((response) => {
              if (response.jsonCode === 200) {
                  Navigation.navigate(ROUTES.SETTINGS_PROFILE);
              }
          });
  }
```

and add `countyCodeRef` ref and focus if country code exist
https://github.com/Expensify/App/blob/f3228deee0ac6a9fabb2c1c5d40ffe31d77dbbd7/src/pages/settings/AddSecondaryLoginPage.js#L109
```
if (this.countyCodeRef) {
    this.countyCodeRef.focus();
} else if (this.phoneNumberInputRef) {
    this.phoneNumberInputRef.focus();
}
```

Here is screenshot
![image](https://user-images.githubusercontent.com/31027036/128935973-56e3cb1b-3c77-4a3d-b45d-9931d46e4bc6.png)

Thanks

**Comment by MitchExpensify** [comment]
Not 100% if this issue relates to this problem but the text in the fields is causing the input to be illegible. 

![image](https://user-images.githubusercontent.com/36425901/129260811-70a4a2a1-5abf-4090-b56a-cfe222055bd8.png)

**Comment by Luke9389** [comment]
Just leaving a quick comment here to shake the `Overdue` label off of this issue. No internal progress on this just yet.

**Comment by Luke9389** [comment]
Just leaving a quick comment here to shake the Overdue label off of this issue. No internal progress on this just yet.

**Comment by Luke9389** [comment]
Looping back around to this soon

**Comment by iwiznia** [comment]
FWIW I am a 👎 on this feature:
> I personally find those selectors annoying. I know my country code already, I rather type it than have to select it from a drop down.

**Comment by sig5** [comment]
Seems like this issue has been inactive for a while, I will drop in my proposal as I would like to work on it.
**Proposal**

**API changes:**
Currently we have an object containing country codes in Onyx. We can instead have a country mapping, which contains country code-specific Details (Here currency type and Country mobile code). Also, we can have country code saved in `myPersonalDetails` instead of `localCurrency` .

_***Client Changes***_
We can introduce a new component AutoCompleteTextInput` which can listen to text changes in TextInput and filter the dropdown suggestions in a view that appears below the input. 
As soon as the dropdown selection is matched, the focus will be shifted to the Phone number `textInput.`

I will post more elaborate details if I can be working on this and it has not been taken up by anyone else.

**Comment by Luke9389** [comment]
Looping back around to this soon

**Comment by Luke9389** [comment]
I'm going to close this for the time being. I don't forsee being able to push this initiative for at least another month.
